Document UUID to object cast (#5176)
edgedb/edgedb: docs/reference/edgeql/casts.rst

See the :eql:op:`type cast operator <cast>` section for more
information on type casting rules.

.. versionadded:: 3.0

    You can cast a UUID into an object:

    .. code-block:: edgeql-repl

        db> select <Hero><uuid>'01d9cc22-b776-11ed-8bef-73f84c7e91e7';
        {default::Hero {id: 01d9cc22-b776-11ed-8bef-73f84c7e91e7}}

>     If you try to cast a UUID that doesn't exist on the object, you'll get an

Reviewer: ```suggestion
    If you try to cast a UUID that no object of the type has, you'll get an
```

Author: Thanks, @1st1. Implemented this manually to observe maximum line lengths.
